commerce-test/src/testIntegration/java/com/liferay/commerce/shipment/test/CommerceShipmentTest.java

```diff
@@ -0,0 +1,1024 @@
+		).when(
+			"I add an item to the order"
+		).then(
+			"The shipping is associated to the order and to the items"
+		);
+
+		CPInstance cpInstance = _createCPInstance();
+
+		CommerceChannel commerceChannel = _createCommerceChannel();
+
+		BigDecimal value = BigDecimal.valueOf(RandomTestUtil.nextDouble());
+
+		CommerceOrder commerceOrder =
+			CommerceTestUtil.createCommerceOrderForShipping(
+				_company.getGroupId(), _user.getUserId(),
+				_commerceCurrency.getCommerceCurrencyId(),
+				commerceChannel.getSiteGroupId(), value);
+
+		_commerceOrders.add(commerceOrder);
+
+		CommerceContext commerceContext = new TestCommerceContext(
+			commerceOrder.getCommerceCurrency(), commerceChannel, null, null,
+			null, commerceOrder);
+
+		CommerceInventoryWarehouse commerceInventoryWarehouse =
+			_createCommerceInventoryWarehouse(
+				commerceChannel.getCommerceChannelId(), true);
+
+		int quantity = 10;
+
+		CommerceInventoryTestUtil.addCommerceInventoryWarehouseItem(
+			_user.getUserId(), commerceInventoryWarehouse, cpInstance.getSku(),
+			quantity);
+
+		int orderedQuantity = 1;
+
+		CommerceOrderItem commerceOrderItem =
+			CommerceTestUtil.addCommerceOrderItem(
+				commerceOrder.getCommerceOrderId(),
+				cpInstance.getCPInstanceId(), orderedQuantity, commerceContext);
+
+		CommerceShipment commerceShipment =
+			CommerceShipmentTestUtil.createOrderShipment(
+				commerceOrder.getGroupId(), commerceOrder.getCommerceOrderId(),
+				commerceInventoryWarehouse.getCommerceInventoryWarehouseId());
+
+		Assert.assertEquals(
+			commerceOrder.getCommerceAccount(),
+			commerceShipment.getCommerceAccount());
+
+		Assert.assertEquals(
+			commerceOrder.getGroupId(), commerceShipment.getGroupId());
+
+		int commerceShipmentItemsCount =
+			_commerceShipmentItemLocalService.getCommerceShipmentItemsCount(
+				commerceShipment.getCommerceShipmentId());
+
+		Assert.assertEquals(orderedQuantity, commerceShipmentItemsCount);
+
+		List<CommerceShipmentItem> commerceShipmentItems =
+			_commerceShipmentItemLocalService.getCommerceShipmentItems(
+				commerceShipment.getCommerceShipmentId(), QueryUtil.ALL_POS,
+				QueryUtil.ALL_POS, null);
+
+		CommerceShipmentItem commerceShipmentItem = commerceShipmentItems.get(
+			0);
+
+		Assert.assertEquals(
+			commerceOrderItem.getCommerceOrderItemId(),
+			commerceShipmentItem.getCommerceOrderItemId());
+
+		Assert.assertEquals(
+			commerceShipment, commerceShipmentItem.getCommerceShipment());
+	}
+
+	@Test
+	public void testCreateOrderWithShippingAmount() throws Exception {
+		frutillaRule.scenario(
+			"When a shipping amount is associated to an order the price " +
+				"object is correctly set"
+		).given(
+			"An order"
+		).and(
+			"A shipping with a shipping amount set"
+		).when(
+			"I attach the shipping to the order"
+		).then(
+			"The price object has the shipping amount correctly set"
+		);
+
+		CPInstance cpInstance = _createCPInstance();
+
+		cpInstance.setPrice(BigDecimal.valueOf(25.0));
```

Comment:
write this as `25` rather than `25.0`. Liferay conventions never have us specify more information than necessary in number literals (e.g. adding `d` or `f` or `.0` to the end). The JVM will just promote the integer as needed.